ui/plugins/pinboard.js

```diff
@@ -158,15 +159,16 @@ treeherder.controller('PinboardCtrl', [
 
         $scope.canSaveClassifications = function() {
             var thisClass = $scope.classification;
-            return $scope.hasPinnedJobs() && (thPinboard.hasRelatedBugs() && $scope.user.loggedin ||
-                   thisClass.failure_classification_id !== 4 && thisClass.failure_classification_id !== 2 ||
+            return $scope.hasPinnedJobs() && $scope.user.loggedin &&
+                   (thPinboard.hasRelatedBugs() ||
+                   (thisClass.failure_classification_id !== 4 && thisClass.failure_classification_id !== 2) ||
```

Comment:
This seems like it's more correctly describing valid configurations than what we had before. Could still probably be improved, though.